### PR TITLE
Pin `protobuf` in cross-version test

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Pin `protobuf` to fix cross-version test failures:
https://github.com/mlflow/mlflow/actions/runs/2400280896

## How is this patch tested?

Existing tests

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/ml-package-versions.yml [hunk 1]
```diff
@@ -122,14 +128,16 @@ xgboost:
   models:
     minimum: "1.1.1"
     maximum: "1.6.1"
-    requirements: ["scikit-learn"]
+    requirements:
+      ">= 0.0.0": ["scikit-learn"]
```
Comment (harupy) [inline mlflow/ml-package-versions.yml:132]: `>= 0.0.0"` means `always install`.

File: mlflow/ml-package-versions.yml [hunk 2]
```diff
@@ -95,9 +98,11 @@ keras:
     minimum: "2.3.0"
     maximum: "2.8.0"
     requirements:
-      "<= 2.3.1": ["tensorflow==2.2.1", "scikit-learn", "pyspark", "pyarrow", "transformers<=4.11.3"]
-      "> 2.3.1, < 2.6.0": ["tensorflow<2.5.0", "scikit-learn", "pyspark", "pyarrow", "transformers"]
-      ">= 2.6.0": ["tensorflow", "scikit-learn", "pyspark", "pyarrow", "transformers"]
+      ">= 0.0.0": ["scikit-learn", "pyspark", "pyarrow"]
+      "<= 2.3.1": ["tensorflow==2.2.1", "transformers<=4.11.3"]
+      "> 2.3.1, < 2.6.0": ["tensorflow<2.5.0", "transformers"]
+      "< 2.6.0": ["protobuf<4.0.0"]
```
Comment (WeichenXu123) [inline mlflow/ml-package-versions.yml:104]: Is this redundant ?
why not writing `"< 4.0.0": ["protobuf"]` ?

File: dev/set_matrix.py [hunk 3]
```diff
@@ -301,24 +258,12 @@ def process_requirements(requirements, version=None):
...
-            )
-            if match_all:
-                return packages
-        return []
+        reqs = set()
+        for specifier, packages in requirements.items():
+            specifier_set = SpecifierSet(specifier.replace(DEV_VERSION, DEV_NUMERIC))
+            if specifier_set.contains(DEV_NUMERIC if version == DEV_VERSION else version):
+                reqs = reqs.union(packages)
+        return sorted(reqs)
```
Comment (WeichenXu123) [inline dev/set_matrix.py:266]: Good simplification!

**Review by WeichenXu123 — COMMENTED** [review]
LGTM, with a minor question.

File: mlflow/ml-package-versions.yml (hunk 2 above)
Comment (harupy) [reply mlflow/ml-package-versions.yml:104]: It's not redundant. `"< 2.6.0": ["protobuf<4.0.0"]` means install `protobuf<4.0.0` if keras version is `< 2.6.0`.

**Review by WeichenXu123 — APPROVED** [review]

**Pull request merged**